### PR TITLE
Added field and class level annotations to the pojo.mustache

### PR DESCRIPTION
**Pull request by jayfray12**
Added x-class-extra-annotation and x-field-extra-annotation for class and field level annotations per object/field for pojo.mustache template for Java Spring.  This fixes issue #11772 

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [ ] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [ ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by jayfray12** [comment]
> Hi @jayfray12 Thanks for your contribution. I have two comments:
> 
@borsch Thanks for the comments.  See responses below
> 1. can you please add this to other pojo.mustache files as well?
Yes I will do this.
> 2. write some tests to verify that this works on per-class & per-field level(it's possible to specify different set of annotations for different classes & different fields inside class)
I tried but wasn't quite sure where to write the tests and the documentation didn't help much either.  I tried to add something to the [SpringCodeGenTest.java](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/java/org/openapitools/codegen/java/spring/SpringCodegenTest.java) class but it didn't seem like it was picking up my new changes to the pojo.mustache file. Can you point me in the right direction here?

**Review by borsch — APPROVED** [review]

**Comment by wing328** [comment]
cc @bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10)

cc @cachescrubber (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02)

**Comment by jayfray12** [comment]
> @jayfray12 thanks for the PR. Looks like the samples are not up-to-date: https://github.com/OpenAPITools/openapi-generator/runs/5451658656?check_suite_focus=true
> 
> Can you please follow step 3 in the PR checklist to update the samples?

@wing328 I tried but the `./mvnw clean package` step fails:
image

And the `bin/generate-samples.sh` step fails as well:
```
[pool-1-thread-5] Generating rust-server (outputs to /home/jfroehli/AppDev/personal/openapi-generator/samples/server/petstore/rust-server/output/no-example-v3)…
[pool-1-thread-9] Generating rust-server (outputs to /home/jfroehli/AppDev/personal/openapi-generator/samples/server/petstore/rust-server/output/multipart-v3)…
[pool-1-thread-11] Generation failed for python-experimental: (RuntimeException) Could not generate model 'AdditionalPropertiesClass'
java.lang.RuntimeException: Could not generate model 'AdditionalPropertiesClass'
	at org.openapitools.codegen.DefaultGenerator.generateModels(DefaultGenerator.java:532)
	at org.openapitools.codegen.DefaultGenerator.generate(DefaultGenerator.java:888)
	at org.openapitools.codegen.cmd.GenerateBatch$GenerationRunner.run(GenerateBatch.java:232)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: com.github.jknack.handlebars.HandlebarsException: model.handlebars:3:3: java.lang.reflect.InaccessibleObjectException: Unable to make field transient java.util.HashMap$Node[] java.util.HashMap.table accessible: module java.base does not "opens java.util" to unnamed module @1760e081
    model.handlebars:3:3
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
	at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:178)
	at java.base/java.lang.reflect.Field.setAccessible(Field.java:172)
	at com.github.jknack.handlebars.context.FieldValueResolver$FieldMember.setAccessible(FieldValueResolver.java:150)
	at com.github.jknack.handlebars.context.MemberValueResolver.cache(MemberValueResolver.java:82)
	at com.github.jknack.handlebars.context.MemberValueResolver.resolve(MemberValueResolver.java:54)
	at com.github.jknack.handlebars.Context$CompositeValueResolver.resolve(Context.java:199)
	at com.github.jknack.handlebars.internal.path.PropertyPath.eval(PropertyPath.java:52)
	at com.github.jknack.handlebars.Context$PathExpressionChain.next(Context.java:361)
	at com.github.jknack.handlebars.Context$PathExpressionChain.eval(Context.java:381)
	at com.github.jknack.handlebars.Context.get(Context.java:618)
	at com.github.jknack.handlebars.internal.Partial.override(Partial.java:253)
	at com.github.jknack.handlebars.internal.Partial.merge(Partial.java:226)
	at com.github.jknack.handlebars.internal.BaseTemplate.apply(BaseTemplate.java:125)
	at com.github.jknack.handlebars.internal.TemplateList.merge(TemplateList.java:95)
	at com.github.jknack.handlebars.internal.BaseTemplate.apply(BaseTemplate.java:125)
	at com.github.jknack.handlebars.internal.BaseTemplate.apply(BaseTemplate.java:113)
	at org.openapitools.codegen.templating.HandlebarsEngineAdapter.compileTemplate(HandlebarsEngineAdapter.java:92)
	at org.openapitools.codegen.TemplateManager.write(TemplateManager.java:163)
	at org.openapitools.codegen.DefaultGenerator.processTemplateToFile(DefaultGenerator.java:1034)
	at org.openapitools.codegen.DefaultGenerator.processTemplateToFile(DefaultGenerator.java:1021)
	at org.openapitools.codegen.DefaultGenerator.generateModel(DefaultGenerator.java:388)
	at org.openapitools.codegen.DefaultGenerator.generateModels(DefaultGenerator.java:523)
	... 5 more
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make field transient java.util.HashMap$Node[] java.util.HashMap.table accessible: module java.base does not "opens java.util" to unnamed module @1760e081
	... 29 more
```

**Comment by borsch** [comment]
@jayfray12 no need to run all tests. you can run `mvn clean && ./bin/utils/ensure-up-to-date`. if you are using Windows then better to run inside docker

**Comment by wing328** [comment]
Can you try adding `-Dmaven.javadoc.skip=true -Dmaven.test.skip=true` to the mvn command?

**Comment by jayfray12** [comment]
Adding `-Dmaven.javadoc.skip=true -Dmaven.test.skip=true` fixed the first step of `mvn clean package` but the second step of the generate sources still fails with the same error

File: modules/openapi-generator/src/main/resources/JavaJaxRS/spec/pojo.mustache [hunk 1]
```diff
@@ -12,12 +12,19 @@ import com.fasterxml.jackson.annotation.JsonTypeName;
  **/{{/description}}
 {{#useSwaggerAnnotations}}{{#description}}@ApiModel(description = "{{{.}}}"){{/description}}{{/useSwaggerAnnotations}}
 @JsonTypeName("{{name}}")
+{{>generatedAnnotation}}{{>additionalModelTypeAnnotations}}
```
Comment (borsch) [inline modules/openapi-generator/src/main/resources/JavaJaxRS/spec/pojo.mustache:15]: remove this duplicated line

**Comment by jayfray12** [comment]
Not sure why the `./CI/circle_parallel.sh` failed.  Output says:
```
Exception when calling PetApi#getPetById
Status code: 404
Reason: {"code":1,"type":"error","message":"Pet not found"}
```
Is there something I can run locally to replicate this to see if this is something my changes caused?

**Comment by jayfray12** [comment]
@wing328 @borsch Looks like my latest update of the samples did the trick and everything is passing now.  I appreciate your help and patience!

**Review by borsch — APPROVED** [review]

**Comment by crux** [comment]
had the same problems, on mac osx. 

`$ brew install openjdk@11`

made it compile.

**Comment by vsnpew** [comment]
@jayfray12 Hello, can i see some example of using these features (x-extra-class-annotation, x-extra-field-annotation) ?

i tried this: 
`components:
  schemas:
   UnitDTO:
      type: object
      x-extra-class-annotation: "@javax.persistence.Id"
      description: some DTO
      properties:
        id:
          type: integer
          format: int64
          description: Id
          example: asd`

But it doenst work.

Do I need to enable some flag in build.gradle?

**Comment by jayfray12** [comment]
@vsnpew Which version are you using.  There isn't a released version with this code in it yet unless you compiled your own.  Sample file with annotations can be found here:
https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/3_0/issue_11772.yml#L72

**Comment by jayfray12** [comment]
It is tagged for the 6.0.0 release which should wrap up by the end of the month
https://github.com/OpenAPITools/openapi-generator/milestone/35

**Comment by bpstelios10** [comment]
is this supposed to work with parameters too? or is it just for schemas?
cause i set a parameter like this and i dont get the not-blank annotation:
```
      name: name
      in: path
      required: true
      description: Unique identifier for the configuration resource
      x-class-extra-annotation: "@jakarta.validation.constraints.NotBlank"
      x-field-extra-annotation: "@jakarta.validation.constraints.NotBlank"
      schema:
        type: string
        minLength: 1
```
(i put both class/field just to make sure)

**Pull request merged**